plugins/google-analytics/readcomplete-event.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * Detects when an article is 25%, 50%, and 100% complete based on scroll position.
+ * The read status of an article is saved in local storage to avoid sending multiple read events to Google Analytics within a single browser session.
+ * Note that in order for this to work, the Google Analytics JS code snippet needs to already be imported. Otherwise, the `gtag()` function will be undefined
+ * 
+ * @param string articleSelector    The DOM article element
+ * @param string articleTitle       The title of the article
+ */
+function detectReadComplete(articleSelector, articleTitle) {
+    // Generate a hash based on the article title to be used for uniquely organizing article progress in local storage
+    const titleHash = (function(str) {
+        let hashString = 0;
+        for (let character of str) {
+           let charCode = character.charCodeAt(0);
+           hashString = hashString << 5 - hashString;
+           hashString += charCode;
+           hashString |= hashString;
+        }
+        return hashString;
+    })(articleTitle);
+
+    // Helper methods for storing and retreiving values about article from local storage
+    const getStoredValue = (key) => localStorage.getItem(titleHash + '-' + key) === 'true';
+    const setStoredValue = (key, value) => localStorage.setItem(titleHash + '-' + key, value ?? false);
```

Comment:
Local Storage can only store string values, but I wanted boolean values; the type conversion between string and boolean values is intentional.